### PR TITLE
ci: clean up workflow files

### PR DESCRIPTION
**Pull request by antoineco**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

Minor clean up in workflow files:

- Let `make` determine the kraftkit VERSION by itself, instead of setting that value manually.
- Do not run Go checks in a container. We already use `setup-go` to pin the Go version.
- Run `actionlint` at the root of the repo without `xargs`. It knows how to find workflows.
- Remove unnecessary semi colons here and there.
- Typos.

### CODE REVIEW

File: .github/workflows/e2e-cli.yaml [hunk 1]
```diff
@@ -17,6 +17,9 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (craciunoiuc) [inline .github/workflows/e2e-cli.yaml:21]: Shouldn't this be `1` to fetch only the last commit? (not sure but maybe 0 means everything)

File: .github/workflows/actcheck.yaml [hunk 2]
```diff
@@ -13,15 +13,11 @@ jobs:
...
-          mkdir -p ~/bin
-          curl -sL https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash | bash -s -- latest ~/bin
+          mkdir -p "$HOME"/.local/bin
+          curl -sL https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash | bash -s -- latest "$HOME"/.local/bin
 
       - name: Check that all workflows are valid
-        run: |
-          set -xe
-          find .github/workflows -type f -name '*.yaml' -print0 | xargs --null ~/bin/actionlint
+        run: actionlint -verbose
```
Comment (craciunoiuc) [inline .github/workflows/actcheck.yaml:23]: Does this autodetect actions? With the previous formatting it was only targetting action yamls

Comment (antoineco) [reply .github/workflows/actcheck.yaml:23]: Yes it does:

```
verbose: Linting all workflow files in repository: .
verbose: Detected project: /home/runner/work/kraftkit/kraftkit
verbose: Collected 11 YAML files
verbose: Linting 11 files
...
(see full output in workflow run)
```

Code location: https://github.com/rhysd/actionlint/blob/v1.6.24/linter.go#L229-L243

File: .github/workflows/e2e-cli.yaml (hunk 1 above)
Comment (antoineco) [reply .github/workflows/e2e-cli.yaml:21]: The default is 1. The problem with shallow clones is that `make` can only populate the VERSION variable based on the most recent _tag_, and a shallow clone doesn't pull enough depth for that tag to exist.

Code location: https://github.com/unikraft/kraftkit/blob/3b3a3029f910418c6b3eb95c0abaad84bf4e647f/Makefile#L24-L25

File: .github/workflows/actcheck.yaml (hunk 2 above)
Comment (craciunoiuc) [reply .github/workflows/actcheck.yaml:23]: Nice, so it detects only the `.github/workflows` file

File: .github/workflows/e2e-cli.yaml (hunk 1 above)
Comment (craciunoiuc) [reply .github/workflows/e2e-cli.yaml:21]: okay

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**